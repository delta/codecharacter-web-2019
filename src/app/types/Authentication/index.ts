export enum AuthType {
  LOGIN = 'LOGIN',
  REGISTER = 'REGISTER',
}

export interface OwnProps {
  setIsAuthenticationOpen: (isAuthenticationOpen: boolean) => void;
}

export interface State {
  authType: AuthType;
}

export interface StateProps {
  isLoggedIn: boolean;
}

export interface DispatchProps {
  updateErrorMessage: (errorMessage: string) => void;
  toggleUserProfileModal: (isUserProfileModalOpen: boolean) => void;
}

export type Props = OwnProps & StateProps & DispatchProps;
