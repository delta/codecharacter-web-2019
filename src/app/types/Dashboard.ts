import { DashboardActions } from 'app/actions';
import { SidePanelTab } from 'app/reducers/Dashboard';
import { ActionType } from 'typesafe-actions';

const actions = {
  resetDashboardState: DashboardActions.resetDashboardState,
  setSidePanelTab: DashboardActions.setSidePanelTab,
};

export enum SplitPaneState {
  EDITOR = 'EDITOR',
  BOTH = 'BOTH',
  RENDERER = 'RENDERER',
}

export interface State {
  editorWidthRatio: number;
  fixedLeftPaneWidth: number;
  rendererHeight: number;
  splitPaneState: SplitPaneState;
  windowWidth: number;
  isJoyRideActive: boolean;
}

export interface StateProps {
  isLoggedIn: boolean;
  sidePanelOpen: boolean;
  isAuthenticationOpen: boolean;
  isUserProfileModalOpen: boolean;
  logFile: string;
  player1DebugLog: string;
  player2DebugLog: string;
}

export interface DispatchProps {
  setIsAuthenticationOpen: (isAuthenticationOpen: boolean) => void;
}

export type Props = StateProps & DispatchProps;

export type DashboardActions = Omit<typeof DashboardActions, 'Type'>;

export interface DashboardStoreState {
  sidePanelTab: SidePanelTab;
}

export type DashboardStoreAction = ActionType<typeof actions>;
