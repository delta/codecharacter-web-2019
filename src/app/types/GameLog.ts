import { GameLogActions } from 'app/actions/GameLog';
import { ActionType } from 'typesafe-actions';

export interface StateProps {
  debugLog: string;
}

export interface ElementProps {
  height: number;
}

export type Props = StateProps & ElementProps;

const actions = {
  clearAllLogs: GameLogActions.clearAllLogs,
  clearDisplayDebugLog: GameLogActions.clearDisplayDebugLog,
  resetGameLog: GameLogActions.resetGameLogState,
  updateDisplayDebugLog: GameLogActions.updateDisplayDebugLog,
  updateGameLog: GameLogActions.updateGameLog,
  updateMatchPlayerId: GameLogActions.updateMatchPlayerId,
};

export interface GameLogStoreState {
  displayDebugLog: string;
  gameLog: string;
  player1DebugLog: string;
  player2DebugLog: string;
  matchPlayerId: number;
}

export type GameLogStoreAction = ActionType<typeof actions>;
