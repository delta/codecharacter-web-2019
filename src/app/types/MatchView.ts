import { MatchActions } from 'app/actions';
import { ActionType } from 'typesafe-actions';

export enum MatchViewTabType {
  MY_MATCHES = 'MY_MATCHES',
  TOP_MATCHES = 'TOP_MATCHES',
}

export interface OwnState {
  activeMatchViewTab: MatchViewTabType;
  offset: number;
}

export interface Match {
  userOne: string;
  userOneScore: number;
  userTwoScore: number;
  userTwo: string;
  winner: string;
  playedAt: Date;
}

export interface StateProps {
  matches: Match[];
  topMatches: Match[];
  loading: boolean;
}

export interface DispatchProps {
  getMatches: () => void;
  getTopMatches: () => void;
}

export type Props = StateProps & DispatchProps;

export type State = OwnState;

export interface ElementOwnProps {
  match: Match;
  index: number;
}

export type ElementProps = ElementOwnProps;

const actions = {
  getMatches: MatchActions.getMatches,
  getTopMatches: MatchActions.getTopMatches,
  updateError: MatchActions.updateError,
  updateLoading: MatchActions.updateLoadingStatus,
  updateMatches: MatchActions.updateMatches,
  updateTopMatchs: MatchActions.updateTopMatches,
};

export interface MatchStoreState {
  loading: boolean;
  matches: Match[];
  topMatches: Match[];
}

export type MatchStoreAction = ActionType<typeof actions>;
