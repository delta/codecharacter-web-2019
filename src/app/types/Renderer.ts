export interface OwnProps {
  height: number;
}

export interface StateProps {
  logFile: string;
  player1DebugLog: string;
  player2DebugLog: string;
}

export interface DispatchProps {
  clearLog: () => void;
  updateLog: (log: string) => void;
}

export type Props = OwnProps & DispatchProps & StateProps;
