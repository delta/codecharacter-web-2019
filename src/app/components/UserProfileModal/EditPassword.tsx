import { faLock, faPen } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import * as styles from 'app/styles/UserProfileModal.module.css';
import { InputName, InputState } from 'app/types/UserProfileModal';
import * as EditPasswordInterfaces from 'app/types/UserProfileModal/EditPassword';
import classnames from 'classnames';
import * as React from 'react';
import { Button, Row } from 'react-bootstrap';

export class EditPassword extends React.Component<EditPasswordInterfaces.Props, {}> {
  public render() {
    const { handleEditPassword, onInputChange, inputEnabler } = this.props;
    const { editPasswordRef } = this.props;
    const { listDisabled, oldPassword, password, repeatPassword } = this.props;
    return (
      <div className="col-6">
        <Row className={classnames('mb-4')}>
          <div className={classnames('col-sm-12', styles.form)}>
            <div
              className={classnames('text-dark', styles.formHeading)}
              style={{
                display: 'flex',
              }}
            >
              {' '}
              Credentials
              <Button
                className={classnames(styles.editPen, 'ml-auto p-0', {
                  [`${styles.editPenActive}`]: !listDisabled.isPasswordDisabled,
                })}
                onClick={() => {
                  inputEnabler(InputState.isPasswordDisabled, !listDisabled.isPasswordDisabled);
                }}
              >
                <FontAwesomeIcon icon={faPen} />
              </Button>
            </div>
            <form
              className={'editpasswordForm'}
              noValidate
              ref={editPasswordRef}
              onSubmit={handleEditPassword}
            >
              <div className="form-row">
                <div className="col mb-4">
                  <div className="input-group">
                    <div className="input-group-prepend">
                      <span className="input-group-text" id="inputGroupPrepend">
                        <FontAwesomeIcon icon={faLock} />
                      </span>
                    </div>
                    <input
                      type="password"
                      className="form-control"
                      id="editValidationOldPassword"
                      placeholder="Old Password"
                      aria-describedby="inputGroupPrepend"
                      minLength={5}
                      value={oldPassword}
                      onChange={(e) => onInputChange(InputName.oldPassword, e.target.value)}
                      required
                      disabled={listDisabled.isPasswordDisabled}
                    />
                    <div className="invalid-feedback">
                      Password should have minimum 5 characters.
                    </div>
                  </div>
                </div>
              </div>
              <div className="form-row">
                <div className="col mb-4">
                  <div className="input-group">
                    <div className="input-group-prepend">
                      <span className="input-group-text" id="inputGroupPrepend">
                        <FontAwesomeIcon icon={faLock} />
                      </span>
                    </div>
                    <input
                      type="password"
                      className="form-control"
                      id="editValidationPassword"
                      placeholder="New Password"
                      aria-describedby="inputGroupPrepend"
                      minLength={5}
                      value={password}
                      onChange={(e) => onInputChange(InputName.password, e.target.value)}
                      required
                      disabled={listDisabled.isPasswordDisabled}
                    />
                    {listDisabled.isPasswordDisabled ? (
                      <div className="invalid-feedback">Passwords should match.</div>
                    ) : null}
                  </div>
                </div>
              </div>
              <div className="form-row">
                <div className="col mb-4">
                  <div className="input-group">
                    <div className="input-group-prepend">
                      <span className="input-group-text" id="inputGroupPrepend">
                        <FontAwesomeIcon icon={faLock} />
                      </span>
                    </div>
                    <input
                      type="password"
                      className="form-control"
                      id="editValidationRepeatPassword"
                      placeholder="Confirm Password"
                      aria-describedby="inputGroupPrepend"
                      minLength={5}
                      value={repeatPassword}
                      onChange={(e) => onInputChange(InputName.repeatPassword, e.target.value)}
                      required
                      disabled={listDisabled.isPasswordDisabled}
                    />
                    {listDisabled.isPasswordDisabled ? (
                      <div className="invalid-feedback">Passwords should match.</div>
                    ) : null}
                  </div>
                </div>
              </div>
              <div
                className="form-row"
                style={{
                  marginTop: '112px',
                }}
              >
                <div className="col-12 text-center">
                  <button
                    className="btn btn-info"
                    type="submit"
                    style={{
                      marginTop: '24px',
                      width: '100%',
                    }}
                  >
                    Edit Credentials
                  </button>
                </div>
              </div>
            </form>
          </div>
        </Row>
      </div>
    );
  }
}
