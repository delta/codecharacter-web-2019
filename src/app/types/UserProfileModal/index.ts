import * as UserInterfaces from 'app/types/User';

export interface StateProps {
  isUserProfileModalOpen: boolean;
  userDetails: UserInterfaces.UserStoreState;
}

export interface ListDisabled {
  isUserNameDisabled?: boolean;
  isFullNameDisabled?: boolean;
  isFlagSelectDisabled?: boolean;
  isPasswordDisabled?: boolean;
}

export enum editFormType {
  OWN_PROFILE = 'OWN_PROFILE',
  OTHER_PROFILE = 'OTHER_PROFILE',
}
export interface State {
  activeForm: editFormType;
  avatar: string;
  username: string;
  oldPassword: string;
  password: string;
  repeatPassword: string;
  country: string;
  fullName: string;
  listDisabled: ListDisabled;
}

export enum InputName {
  avatar = 'avatar',
  username = 'username',
  oldPassword = 'oldPassword',
  password = 'password',
  repeatPassword = 'repeatPassword',
  country = 'country',
  fullName = 'fullName',
  listDisabled = 'listDisabled',
}

export enum InputState {
  isUserNameDisabled = 'isUserNameDisabled',
  isFullNameDisabled = 'isFullNameDisabled',
  isFlagSelectDisabled = 'isFlagSelectDisabled',
  isPasswordDisabled = 'isPasswordDisabled',
}

export interface DispatchProps {
  checkUsernameExists: (username: string) => void;
  editUserPassword: (editUserPasswordDetails: UserInterfaces.EditUserPassword) => void;
  editUserProfile: (editUserDetails: UserInterfaces.EditUserDetails) => void;
  getUserDetails: () => void;
  toggleUserProfileModal: (isUserProfileModalOpen: boolean) => void;
}

export type Props = StateProps & DispatchProps;
