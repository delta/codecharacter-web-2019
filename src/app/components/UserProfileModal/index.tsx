import { faTimes } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { EditPassword } from 'app/components/UserProfileModal/EditPassword';
import { EditProfile } from 'app/components/UserProfileModal/EditProfile';
import * as styles from 'app/styles/UserProfileModal.module.css';
import * as UserProfileInterfaces from 'app/types/UserProfileModal';
import classnames from 'classnames';
import * as React from 'react';
import { Button, Grid, Row } from 'react-bootstrap';
// tslint:disable-next-line:import-name
import ReactFlagsSelect from 'react-flags-select';
export class UserProfileModal extends React.Component<
  UserProfileInterfaces.Props,
  UserProfileInterfaces.State
> {
  private editProfileRef = React.createRef<HTMLFormElement>();
  private editPasswordRef = React.createRef<HTMLFormElement>();
  private reactFlagRef = React.createRef<ReactFlagsSelect>();

  constructor(props: UserProfileInterfaces.Props) {
    super(props);
    const { userDetails } = this.props;
    this.state = {
      avatar: userDetails.country,
      country: userDetails.country,
      fullName: userDetails.fullName,
      listDisabled: {
        isFlagSelectDisabled: true,
        isFullNameDisabled: true,
        isPasswordDisabled: true,
        isUserNameDisabled: true,
      },
      oldPassword: '',
      password: '',
      repeatPassword: '',
      username: userDetails.username,
    };
    this.props.getUserDetails();
  }

  public componentWillReceiveProps(nextProps: UserProfileInterfaces.Props) {
    const { country, username, fullName, avatar } = this.state;
    const { userDetails } = nextProps;
    this.setState({
      avatar: userDetails.avatar !== avatar ? userDetails.avatar : avatar,
      country: userDetails.country !== country ? userDetails.country : country,
      fullName: userDetails.fullName !== fullName ? userDetails.fullName : fullName,
      username: userDetails.username !== country ? username : userDetails.username,
    });
    if (this.reactFlagRef.current) {
      this.reactFlagRef.current.updateSelected(
        userDetails.country !== country ? userDetails.country : country,
      );
    }
  }

  public render() {
    const {
      fullName,
      username,
      country,
      oldPassword,
      password,
      repeatPassword,
      listDisabled,
      avatar,
    } = this.state;
    const { isUserProfileModalOpen, toggleUserProfileModal, userDetails } = this.props;
    return (
      <div
        className={classnames(styles.UserProfileModal, {
          [`${styles.modalOpen}`]: isUserProfileModalOpen,
        })}
      >
        <Grid
          fluid={true}
          className={classnames(styles.modal, {
            [`${styles.slideEnter}`]: isUserProfileModalOpen,
          })}
        >
          <Row>
            <div className="col" style={{ padding: 10 }}>
              <div className="col text-center mb-1">
                <h4 className="text-dark m-2">Edit Profile</h4>
              </div>
            </div>
            <div className="col-sm-auto p-0 m-0">
              <Button
                className={classnames(styles.customBtn)}
                onClick={() => {
                  toggleUserProfileModal(false);
                }}
              >
                <FontAwesomeIcon icon={faTimes} color={'black'} />
              </Button>
            </div>
          </Row>
          {/* <Row>
            <div className="col text-center mb-3">
              <h3 className="text-dark m-0">Edit Profile</h3>
            </div>
          </Row> */}
          <div className="text-center errorMessage">{userDetails.errorMessage}</div>
          <Row>
            <EditProfile
              handleEditProfile={this.handleEditProfile}
              onInputChange={this.onInputChange}
              inputEnabler={this.inputEnabler}
              editProfileRef={this.editProfileRef}
              reactFlagRef={this.reactFlagRef}
              username={username}
              listDisabled={listDisabled}
              fullName={fullName}
              userDetails={userDetails}
              country={country}
              avatar={avatar}
            />
            <EditPassword
              handleEditPassword={this.handleEditPassword}
              onInputChange={this.onInputChange}
              inputEnabler={this.inputEnabler}
              editPasswordRef={this.editPasswordRef}
              listDisabled={listDisabled}
              oldPassword={oldPassword}
              password={password}
              repeatPassword={repeatPassword}
              userDetails={userDetails}
            />
          </Row>
        </Grid>
      </div>
    );
  }

  private handleEditProfile = (event: React.FormEvent<HTMLFormElement>) => {
    const { editUserProfile } = this.props;
    const { country, fullName, username, listDisabled, avatar } = this.state;
    const form = this.editProfileRef.current;
    event.preventDefault();

    if (form) {
      if (form.checkValidity()) {
        editUserProfile({
          ...(!listDisabled.isFlagSelectDisabled && { country }),
          ...(!listDisabled.isFullNameDisabled && { fullName }),
          ...(!listDisabled.isUserNameDisabled && { username }),
          avatar,
        });

        this.setState({
          listDisabled: {
            ...listDisabled,
            isEmailDisabled: true,
            isFlagSelectDisabled: true,
            isFullNameDisabled: true,
            isUserNameDisabled: true,
          },
        });
      }
      form.classList.add('was-validated');
    }
  };

  private handleEditPassword = (event: React.FormEvent<HTMLFormElement>) => {
    const { editUserPassword } = this.props;
    const { oldPassword, repeatPassword, password, listDisabled } = this.state;
    const form = this.editPasswordRef.current;
    event.preventDefault();

    if (form) {
      if (form.checkValidity()) {
        if (!listDisabled.isPasswordDisabled && password === repeatPassword) {
          editUserPassword({
            ...(!listDisabled.isPasswordDisabled && {
              oldPassword,
              password,
            }),
          });
        }
        this.setState({
          listDisabled: {
            ...listDisabled,
            isPasswordDisabled: true,
          },
        });
      }
      form.classList.add('was-validated');
    }
  };

  private onInputChange = (inputName: UserProfileInterfaces.InputName, value: string) => {
    if (inputName === UserProfileInterfaces.InputName.username) {
      const { checkUsernameExists } = this.props;
      checkUsernameExists(value);
    }
    this.setState(this.updateInputState(inputName, value));
  };

  private inputEnabler = (inputState: UserProfileInterfaces.InputState, value: boolean) => {
    this.setState(this.updateListDisabledState(inputState, value));
  };

  private updateInputState = <T extends string>(
    key: keyof UserProfileInterfaces.State,
    value: T,
  ) => (prevState: UserProfileInterfaces.State): UserProfileInterfaces.State => ({
    ...prevState,
    [key]: value,
  });

  private updateListDisabledState = <T extends boolean>(
    key: keyof UserProfileInterfaces.ListDisabled,
    value: T,
  ) => (prevState: UserProfileInterfaces.State): UserProfileInterfaces.State => ({
    ...prevState,
    listDisabled: {
      ...prevState.listDisabled,
      [key]: value,
    },
  });
}
