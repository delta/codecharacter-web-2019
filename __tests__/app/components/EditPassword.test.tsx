import { EditPassword } from 'app/components/UserProfileModal/EditPassword';
import { InputName, InputState } from 'app/types/UserProfileModal';
import { mount } from 'enzyme';
import * as React from 'react';

describe('EditPassword Component', () => {
  const wrapper = mount(
    <EditPassword
      handleEditPassword={(e: React.FormEvent<HTMLFormElement>) => {
        return;
      }}
      onInputChange={(inputName: InputName, value: string) => {
        return;
      }}
      inputEnabler={(inputState: InputState, value: boolean) => {
        return;
      }}
      editPasswordRef={React.createRef<HTMLFormElement>()}
      listDisabled={{}}
      oldPassword={'oldPassword'}
      password={'password'}
      repeatPassword={'repeatPassword'}
      userDetails={{
        avatar: '',
        college: '',
        country: 'IN',
        email: '',
        errorMessage: '',
        fullName: '',
        isLoggedIn: true,
        isUserProfileModalOpen: false,
        type: '',
        username: '',
      }}
    />,
  );

  it('Should render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
