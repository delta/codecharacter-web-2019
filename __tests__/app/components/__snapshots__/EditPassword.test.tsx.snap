// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditPassword Component Should render 1`] = `
<EditPassword
  editPasswordRef={
    Object {
      "current": <form
        class="editpasswordForm"
        novalidate=""
      >
        <div
          class="form-row"
        >
          <div
            class="col mb-3"
          >
            <div
              class="input-group"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text"
                  id="inputGroupPrepend"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-lock fa-w-14 "
                    data-icon="lock"
                    data-prefix="fas"
                    role="img"
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </div>
              <input
                aria-describedby="inputGroupPrepend"
                class="form-control"
                id="editValidationOldPassword"
                minlength="5"
                placeholder="Old Password"
                required=""
                type="password"
                value="oldPassword"
              />
              <div
                class="invalid-feedback"
              >
                Password should have minimum 5 characters.
              </div>
            </div>
          </div>
        </div>
        <div
          class="form-row"
        >
          <div
            class="col mb-3"
          >
            <div
              class="input-group"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text"
                  id="inputGroupPrepend"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-lock fa-w-14 "
                    data-icon="lock"
                    data-prefix="fas"
                    role="img"
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </div>
              <input
                aria-describedby="inputGroupPrepend"
                class="form-control"
                id="editValidationPassword"
                minlength="5"
                placeholder="New Password"
                required=""
                type="password"
                value="password"
              />
            </div>
          </div>
        </div>
        <div
          class="form-row"
        >
          <div
            class="col mb-3"
          >
            <div
              class="input-group"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text"
                  id="inputGroupPrepend"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-lock fa-w-14 "
                    data-icon="lock"
                    data-prefix="fas"
                    role="img"
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </div>
              <input
                aria-describedby="inputGroupPrepend"
                class="form-control"
                id="editValidationRepeatPassword"
                minlength="5"
                placeholder="Confirm Password"
                required=""
                type="password"
                value="repeatPassword"
              />
            </div>
          </div>
        </div>
        <div
          class="form-row"
        >
          <div
            class="col-12 text-center"
          >
            <button
              class="btn btn-success"
              style="margin-bottom: 10px; width: 100%;"
              type="submit"
            >
              Save Changes
            </button>
          </div>
        </div>
      </form>,
    }
  }
  handleEditPassword={[Function]}
  inputEnabler={[Function]}
  listDisabled={Object {}}
  oldPassword="oldPassword"
  onInputChange={[Function]}
  password="password"
  repeatPassword="repeatPassword"
  userDetails={
    Object {
      "avatar": "",
      "college": "",
      "country": "IN",
      "email": "",
      "errorMessage": "",
      "fullName": "",
      "isAuthenticationOpen": true,
      "isLoggedIn": true,
      "isUserProfileModalOpen": false,
      "type": "",
      "username": "",
    }
  }
>
  <div
    className="col-12"
  >
    <Row
      bsClass="row"
      className="mb-3"
      componentClass="div"
    >
      <div
        className="mb-3 row"
      >
        <div
          className="col-sm-12"
        >
          <div
            className="text-light"
            style={
              Object {
                "display": "flex",
              }
            }
          >
             
            Credentials
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="default"
              className="ml-auto p-0 undefined"
              disabled={false}
              onClick={[Function]}
            >
              <button
                className="ml-auto p-0 undefined btn btn-default"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        512,
                        512,
                        Array [],
                        "f304",
                        "M290.74 93.24l128.02 128.02-277.99 277.99-114.14 12.6C11.35 513.54-1.56 500.62.14 485.34l12.7-114.22 277.9-277.88zm207.2-19.06l-60.11-60.11c-18.75-18.75-49.16-18.75-67.91 0l-56.55 56.55 128.02 128.02 56.55-56.55c18.75-18.76 18.75-49.16 0-67.91z",
                      ],
                      "iconName": "pen",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-pen fa-w-16 "
                    data-icon="pen"
                    data-prefix="fas"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M290.74 93.24l128.02 128.02-277.99 277.99-114.14 12.6C11.35 513.54-1.56 500.62.14 485.34l12.7-114.22 277.9-277.88zm207.2-19.06l-60.11-60.11c-18.75-18.75-49.16-18.75-67.91 0l-56.55 56.55 128.02 128.02 56.55-56.55c18.75-18.76 18.75-49.16 0-67.91z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </FontAwesomeIcon>
              </button>
            </Button>
          </div>
          <form
            className="editpasswordForm"
            noValidate={true}
            onSubmit={[Function]}
          >
            <div
              className="form-row"
            >
              <div
                className="col mb-3"
              >
                <div
                  className="input-group"
                >
                  <div
                    className="input-group-prepend"
                  >
                    <span
                      className="input-group-text"
                      id="inputGroupPrepend"
                    >
                      <FontAwesomeIcon
                        border={false}
                        className=""
                        fixedWidth={false}
                        flip={null}
                        icon={
                          Object {
                            "icon": Array [
                              448,
                              512,
                              Array [],
                              "f023",
                              "M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z",
                            ],
                            "iconName": "lock",
                            "prefix": "fas",
                          }
                        }
                        inverse={false}
                        listItem={false}
                        mask={null}
                        pull={null}
                        pulse={false}
                        rotation={null}
                        size={null}
                        spin={false}
                        symbol={false}
                        title=""
                        transform={null}
                      >
                        <svg
                          aria-hidden="true"
                          className="svg-inline--fa fa-lock fa-w-14 "
                          data-icon="lock"
                          data-prefix="fas"
                          role="img"
                          style={Object {}}
                          viewBox="0 0 448 512"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z"
                            fill="currentColor"
                            style={Object {}}
                          />
                        </svg>
                      </FontAwesomeIcon>
                    </span>
                  </div>
                  <input
                    aria-describedby="inputGroupPrepend"
                    className="form-control"
                    id="editValidationOldPassword"
                    minLength={5}
                    onChange={[Function]}
                    placeholder="Old Password"
                    required={true}
                    type="password"
                    value="oldPassword"
                  />
                  <div
                    className="invalid-feedback"
                  >
                    Password should have minimum 5 characters.
                  </div>
                </div>
              </div>
            </div>
            <div
              className="form-row"
            >
              <div
                className="col mb-3"
              >
                <div
                  className="input-group"
                >
                  <div
                    className="input-group-prepend"
                  >
                    <span
                      className="input-group-text"
                      id="inputGroupPrepend"
                    >
                      <FontAwesomeIcon
                        border={false}
                        className=""
                        fixedWidth={false}
                        flip={null}
                        icon={
                          Object {
                            "icon": Array [
                              448,
                              512,
                              Array [],
                              "f023",
                              "M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z",
                            ],
                            "iconName": "lock",
                            "prefix": "fas",
                          }
                        }
                        inverse={false}
                        listItem={false}
                        mask={null}
                        pull={null}
                        pulse={false}
                        rotation={null}
                        size={null}
                        spin={false}
                        symbol={false}
                        title=""
                        transform={null}
                      >
                        <svg
                          aria-hidden="true"
                          className="svg-inline--fa fa-lock fa-w-14 "
                          data-icon="lock"
                          data-prefix="fas"
                          role="img"
                          style={Object {}}
                          viewBox="0 0 448 512"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z"
                            fill="currentColor"
                            style={Object {}}
                          />
                        </svg>
                      </FontAwesomeIcon>
                    </span>
                  </div>
                  <input
                    aria-describedby="inputGroupPrepend"
                    className="form-control"
                    id="editValidationPassword"
                    minLength={5}
                    onChange={[Function]}
                    placeholder="New Password"
                    required={true}
                    type="password"
                    value="password"
                  />
                </div>
              </div>
            </div>
            <div
              className="form-row"
            >
              <div
                className="col mb-3"
              >
                <div
                  className="input-group"
                >
                  <div
                    className="input-group-prepend"
                  >
                    <span
                      className="input-group-text"
                      id="inputGroupPrepend"
                    >
                      <FontAwesomeIcon
                        border={false}
                        className=""
                        fixedWidth={false}
                        flip={null}
                        icon={
                          Object {
                            "icon": Array [
                              448,
                              512,
                              Array [],
                              "f023",
                              "M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z",
                            ],
                            "iconName": "lock",
                            "prefix": "fas",
                          }
                        }
                        inverse={false}
                        listItem={false}
                        mask={null}
                        pull={null}
                        pulse={false}
                        rotation={null}
                        size={null}
                        spin={false}
                        symbol={false}
                        title=""
                        transform={null}
                      >
                        <svg
                          aria-hidden="true"
                          className="svg-inline--fa fa-lock fa-w-14 "
                          data-icon="lock"
                          data-prefix="fas"
                          role="img"
                          style={Object {}}
                          viewBox="0 0 448 512"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z"
                            fill="currentColor"
                            style={Object {}}
                          />
                        </svg>
                      </FontAwesomeIcon>
                    </span>
                  </div>
                  <input
                    aria-describedby="inputGroupPrepend"
                    className="form-control"
                    id="editValidationRepeatPassword"
                    minLength={5}
                    onChange={[Function]}
                    placeholder="Confirm Password"
                    required={true}
                    type="password"
                    value="repeatPassword"
                  />
                </div>
              </div>
            </div>
            <div
              className="form-row"
            >
              <div
                className="col-12 text-center"
              >
                <button
                  className="btn btn-success"
                  style={
                    Object {
                      "marginBottom": "10px",
                      "width": "100%",
                    }
                  }
                  type="submit"
                >
                  Save Changes
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </Row>
  </div>
</EditPassword>
`;
